jpegenc/subblocks/dct_1d/dct_1d.py

```diff
@@ -128,23 +128,23 @@ def coeff_assign():
     def mul_add():
         """multiplication and addition"""
         if input_interface.data_valid:
-                mult_reg[0].next = data_in_reg * coeffs[0]
-                mult_reg[1].next = data_in_reg * coeffs[1]
-                mult_reg[2].next = data_in_reg * coeffs[2]
-                mult_reg[3].next = data_in_reg * coeffs[3]
-                mult_reg[4].next = data_in_reg * coeffs[4]
-                mult_reg[5].next = data_in_reg * coeffs[5]
-                mult_reg[6].next = data_in_reg * coeffs[6]
-                mult_reg[7].next = data_in_reg * coeffs[7]
-
-                adder_reg[0].next = mux_flush[0] + mult_reg[0]
-                adder_reg[1].next = mux_flush[1] + mult_reg[1]
-                adder_reg[2].next = mux_flush[2] + mult_reg[2]
-                adder_reg[3].next = mux_flush[3] + mult_reg[3]
-                adder_reg[4].next = mux_flush[4] + mult_reg[4]
-                adder_reg[5].next = mux_flush[5] + mult_reg[5]
-                adder_reg[6].next = mux_flush[6] + mult_reg[6]
-                adder_reg[7].next = mux_flush[7] + mult_reg[7]
+            mult_reg[0].next = data_in_reg * coeffs[0]
```

Comment:
Here is an example where DSP48 (`mac` in the figure) is inferred fine from a loop.  This shows the percentage of resources used on a spartan 6 for a pipelined parallel FIR filter.  It is implemented with loops (no way in hell I would gather all this data having to manually change the code for the filter sizes).

![dsp48_usage_example](https://cloud.githubusercontent.com/assets/766391/16303530/b13f0050-3916-11e6-820d-ef32257cd90a.png)


The inference depends less if a loop is used or not but more on the number of registers before and after the operation.  Being fairly familiar with the synthesis guidelines I don't recall any limitation outlined and I didn't see them in the pages you referred.  What am I missing?

In this project we should be able to generate similar data, for the transform implementation what are the resource utilization for different settings.